### DISCUSSION TITLE
supporting native pipewire

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#5129] [Platforms]
there is an alternative audio engine that everyone on linux are switching to, which is pipewire, it aims to be a drop-in replacement for both pulseaudio and JACK, its even the default audio engine for distros such as fedora

From the official site https://pipewire.org/ :
[PipeWire](https://gitlab.freedesktop.org/pipewire/pipewire) is a project that aims to greatly improve handling of audio and video under Linux. It provides a low-latency, graph based processing engine on top of audio and video devices that can be used to support the use cases currently handled by both pulseaudio and JACK. PipeWire was designed with a powerful security model that makes interacting with audio and video devices from containerized applications easy, with supporting Flatpak applications being the primary goal. Alongside Wayland and Flatpak we expect PipeWire to provide a core building block for the future of Linux application development.

    Capture and playback of audio and video with minimal latency.
    Real-time Multimedia processing on audio and video.
    Multiprocess architecture to let applications share multimedia content.
    Seamless support for PulseAudio, JACK, ALSA and GStreamer applications.
    Sandboxed applications support.

**Comment by Calinou** [comment]
Godot can already use PipeWire via PulseAudio (or even ALSA) – I'm using it right now on Fedora 36.

As I understand it, this is how client applications are expected to use PipeWire. [They don't actually talk to PipeWire directly](https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/FAQ#what-audio-api-do-you-recommend-to-use), but they talk to PulseAudio which has a bridge to PipeWire.

As for native JACK support, someone contributed a driver for it a while ago but it wasn't merged: https://github.com/godotengine/godot/pull/35613
